server/src/internalClusterTest/java/org/opensearch/remotestore/PrimaryTermValidationIT.java

```diff
@@ -0,0 +1,165 @@
+        logger.info("Node names : clusterManager={} primary={} replica={}", clusterManagerNode, primaryNode, replicaNode);
+
+        // Index some docs and validate that both primary and replica node has it. Refresh is triggered to
+        int numOfDocs = randomIntBetween(5, 10);
+        for (int i = 0; i < numOfDocs; i++) {
+            indexSameDoc(clusterManagerNode, INDEX_NAME);
+        }
+        refresh(INDEX_NAME);
+        assertBusy(
+            () -> assertHitCount(client(primaryNode).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), numOfDocs)
+        );
+        assertBusy(
+            () -> assertHitCount(client(replicaNode).prepareSearch(INDEX_NAME).setSize(0).setPreference("_only_local").get(), numOfDocs)
+        );
+
+        // Start network disruption - primary node will be isolated
+        Set<String> nodesInOneSide = Stream.of(clusterManagerNode, replicaNode).collect(Collectors.toCollection(HashSet::new));
+        Set<String> nodesInOtherSide = Stream.of(primaryNode).collect(Collectors.toCollection(HashSet::new));
+        NetworkDisruption networkDisruption = new NetworkDisruption(
+            new NetworkDisruption.TwoPartitions(nodesInOneSide, nodesInOtherSide),
+            NetworkDisruption.DISCONNECT
+        );
+        internalCluster().setDisruptionScheme(networkDisruption);
+        logger.info("--> network disruption is started");
+        networkDisruption.startDisrupting();
+
+        // Ensure the node which is partitioned is removed from the cluster
+        assertBusy(() -> {
+            NodesInfoResponse response = client(clusterManagerNode).admin().cluster().prepareNodesInfo().get();
+            assertThat(response.getNodes().size(), equalTo(2));
+        });
+
+        // Ensure that the cluster manager has latest information about the index
+        assertBusy(() -> {
+            ClusterHealthResponse clusterHealthResponse = client(clusterManagerNode).admin()
+                .cluster()
+                .health(new ClusterHealthRequest())
+                .actionGet(TimeValue.timeValueSeconds(1));
+            assertTrue(clusterHealthResponse.getIndices().containsKey(INDEX_NAME));
+            ClusterIndexHealth clusterIndexHealth = clusterHealthResponse.getIndices().get(INDEX_NAME);
+            assertEquals(ClusterHealthStatus.YELLOW, clusterHealthResponse.getStatus());
+            assertEquals(1, clusterIndexHealth.getNumberOfShards());
+            assertEquals(1, clusterIndexHealth.getActiveShards());
+            assertEquals(1, clusterIndexHealth.getUnassignedShards());
+            assertEquals(1, clusterIndexHealth.getUnassignedShards());
+            assertEquals(1, clusterIndexHealth.getActivePrimaryShards());
+            assertEquals(ClusterHealthStatus.YELLOW, clusterIndexHealth.getStatus());
+        });
+
+        // Index data to the newly promoted primary
+        indexSameDoc(clusterManagerNode, INDEX_NAME);
```

Comment:
`replicaNode` would be newly promoted primary, right?

Reply:
So, clusterManager acts as a co-ordinator here, right?

Reply:
Yes